### PR TITLE
Stopped views when they are idle

### PR DESCRIPTION
**Pull request by umbreak**
Fixes https://github.com/BlueBrain/nexus/issues/2461
Fixes https://github.com/BlueBrain/nexus/issues/2518

TODO: 
- https://github.com/BlueBrain/nexus/issues/2519

### CODE REVIEW

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphOnEventInstant.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    config: BlazegraphViewsConfig,
+    coordinator: BlazegraphIndexingCoordinator
+) extends OnEventInstant {
+
+  override def awakeIndexingStream(
+      project: ProjectRef,
+      prevEvent: Option[Instant],
+      currentEvent: Instant
+  ): Task[Unit] = {
+    val idleTimeout = config.idleTimeout.minus(config.idleTimeout.div(10).min(1.minute)) // allow for some tolerance
```
Comment (imsdu) [inline delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphOnEventInstant.scala:27]: How did you come up with this formula ? Substracting a minute is not enough ?

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphOnEventInstant.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    Task.when(currentEvent.diff(prevEvent.getOrElse(currentEvent)).gteq(idleTimeout)) {
+      val currentProjectViews = views.list(OnePage, searchParams(project), ResourceF.defaultSort).map(_.sources)
```
Comment (imsdu) [inline delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphOnEventInstant.scala:29]: You will get aggregate views here too no ?

File: delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchOnEventInstant.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+) extends OnEventInstant {
+
+  override def awakeIndexingStream(
+      project: ProjectRef,
+      prevEvent: Option[Instant],
+      currentEvent: Instant
+  ): Task[Unit] = {
+    val idleTimeout = config.idleTimeout.minus(config.idleTimeout.div(10).min(1.minute)) // allow for some tolerance
+    Task.when(currentEvent.diff(prevEvent.getOrElse(currentEvent)).gteq(idleTimeout)) {
+      val currentProjectViews = views.list(OnePage, searchParams(project), ResourceF.defaultSort).map(_.sources)
```
Comment (imsdu) [inline delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchOnEventInstant.scala:30]: Same, you will get aggregate views no ?

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/sdk-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/views/indexing/IndexingStreamAwake.scala [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+              onEvent
+                .awakeIndexingStream(evProject, acc.value.value.get(evProject), evInstant)
+                .as(msg.as(ProjectsEventsInstantCollection(acc.value.value + (evProject -> evInstant))))
+            }
+            .persistProgress(progress, projectionId, projection, persistProgressConfig)
+            .void
+        }
+
+    val retryStrategy =
+      RetryStrategy[Throwable](keyValueStoreConfig.retry, _ => true, logError(logger, "IndexingStreamAwake"))
```
[the remark this answers is not in the document]
Comment (umbreak) [reply delta/sdk-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/views/indexing/IndexingStreamAwake.scala:79]: I think It is alright to have the same configuration than for `ProjectsCounts` for now. Let's see later on.

**Review by wwajerowicz — APPROVED** [review]

**Review by wwajerowicz — APPROVED** [review]

**Review by imsdu — APPROVED** [review]

**Pull request merged**